### PR TITLE
Feature/careguide

### PR DESCRIPTION
**Pull request by Yaar-Nuul**
# Description
This commit updates the unit tests to ensure compatibility with recent changes and adds additional coverage for edge cases.

Please include a summary of the change and which issue is fixed. 
Please also include relevant motivation and context. 
List any dependencies that are required for this change.

Perhaps answering the following questions will help create a good description:
* What is this?
This commit includes updates and additions to the unit tests to verify the recent code changes and ensure robustness across various scenarios.

* Why this?
Recent changes to the codebase introduced new features and bug fixes that required corresponding unit tests. This update ensures that all new functionality is properly tested and that existing functionality remains stable.

* How are you doing this? 
They updated existing unit tests to reflect changes in the codebase.
We added new unit tests to cover edge cases and new features.
It is ensured that all tests run without errors and pass successfully.

Fixes # (issue) [Provide a link to the related Jira Task]

## Type of change

Specify the kind of PR/MR

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?
 Verified that all existing unit tests pass successfully.
 Added and tested new unit tests for recently implemented features and edge cases.

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by Minu-Ochieng — APPROVED** [review]

**Review by Bridget-Kathure — APPROVED** [review]

**Review by shallo-nairat — APPROVED** [review]

**Review by Shalom-uwamahoro — APPROVED** [review]

File: mamamind/api/serializers.py [hunk 1]
```diff
@@ -135,24 +132,44 @@ class EPDSQuestionSerializer(serializers.ModelSerializer):
...
+               
 
 class ScreeningTestScoreSerializer(serializers.ModelSerializer):
     class Meta:
         model = ScreeningTestScore
-        fields = '__all__'
+        fields = ['test_date', 'total_score', 'mother_id', 'chp_id']  # Include all required fields here
+        
+
+    def validate(self, data):
```
Comment (Ndichu-shee) [inline mamamind/api/serializers.py:143]: remove this, the serializer is already doing this

**Review by Ndichu-shee — APPROVED** [review]
